Code/Engine/Core/World/Implementation/World.cpp

```diff
@@ -111,6 +112,9 @@ void ezWorld::Clear()
   {
     DeleteObjectNow(it->GetHandle());
   }
+
+  // make sure all dead objects and components are cleared right now
+  Update();
```

Comment:
This is dangerous since it will call all the update logic which can have whatever side effects. If you really need this here you should only call DeleteDeadObjects(); and DeleteDeadComponents(); directly.

Reply:
done